### ISSUE TITLE
quality of text diffuser-2

### ISSUE THREAD

**Issue by lwb2099** [microsoft/unilm#1416]
**Describe**
Model I am using (Text diffuser-2):
I am running inference on text diffuser-2 , the inference code of mine:
`CUDA_VISIBLE_DEVICES=6 python inference_textdiffuser2_t2i_full.py \
  --pretrained_model_name_or_path="/path/to/stable-diffusion-v1-5" \
  --mixed_precision="fp16" \
  --enable_xformers_memory_efficient_attention \
  --resume_from_checkpoint="/path/to/textdiffuser-2" \
  --granularity=128 \
  --max_length=77 \
  --coord_mode="ltrb" \
  --cfg=7.5 \
  --sample_steps=20 \
  --seed=43555 \
  --vis_num 16 \
  --m1_model_path="/path/to/layout_planner" \
  --input_format='prompt' \
  --input_prompt 'A picture of a bruised apple with the text apples are good for you' \
  --output_dir="." `
and the generated results looks like this:
![0](https://github.com/microsoft/unilm/assets/75159534/cd754ee7-c437-4b1b-bd64-cbb25ede15e8)
Looks like something is going wrong. 
Further test on some data from MarioEval:
<img width="436" alt="image" src="https://github.com/microsoft/unilm/assets/75159534/a3b813cc-7672-40b1-9e8e-6be6171c5c0d">
<img width="426" alt="image" src="https://github.com/microsoft/unilm/assets/75159534/db00b972-847d-4256-923f-6d0dc1aeda3a">

**Comment by JingyeChen** [comment]
Did you specify the path of the checkpoints?

**Comment by haberchr** [comment]
@JingyeChen Are there any checkpoints of the TextDiffuser**_-2_** models available based on SD 2.1? 
If not, are there significant modifications to the code required to support the higher-resolution SD model? And, if so, is the training code to support SD 2.1 training released?

**Comment by nosalorenz** [comment]
I would love to test a checkpoint based on SD 2.1 too. The paper already mention that results basing in SD 2.1 are better.

**Comment by 5RJ** [comment]
I meet the similar results, this is my running code, and results as follows:
export CUDA_VISIBLE_DEVICES=4
accelerate launch inference_textdiffuser2_t2i_full.py \
  --pretrained_model_name_or_path="/home/jovyan/wrj/workspace/project/tools/stable-diffusion-v1-5" \
  --mixed_precision="fp16" \
  --output_dir="inference_results" \
  --enable_xformers_memory_efficient_attention \
  --resume_from_checkpoint="/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2-full-ft" \
  --granularity=128 \
  --max_length=77 \
  --coord_mode="ltrb" \
  --cfg=7.5 \
  --sample_steps=20 \
  --seed=43555 \
  --m1_model_path="/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2_layout_planner" \
  --input_format='prompt' \
  --input_prompt='the log for "ABC"'
![image](https://github.com/microsoft/unilm/assets/41320257/3615089b-51ad-4fd2-bd87-d6b7b0bf83dc)
 Does it work normally?

**Comment by 5RJ** [comment]
> I meet the similar results, this is my running code, and results as follows: export CUDA_VISIBLE_DEVICES=4 accelerate launch inference_textdiffuser2_t2i_full.py --pretrained_model_name_or_path="/home/jovyan/wrj/workspace/project/tools/stable-diffusion-v1-5" --mixed_precision="fp16" --output_dir="inference_results" --enable_xformers_memory_efficient_attention --resume_from_checkpoint="/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2-full-ft" --granularity=128 --max_length=77 --coord_mode="ltrb" --cfg=7.5 --sample_steps=20 --seed=43555 --m1_model_path="/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2_layout_planner" --input_format='prompt' --input_prompt='the log for "ABC"' ![image](https://private-user-images.githubusercontent.com/41320257/335524371-3615089b-51ad-4fd2-bd87-d6b7b0bf83dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNDgwMjcsIm5iZiI6MTcxNzE0NzcyNywicGF0aCI6Ii80MTMyMDI1Ny8zMzU1MjQzNzEtMzYxNTA4OWItNTFhZC00ZmQyLWJkODctZDZiN2IwYmY4M2RjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDA5Mjg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2NzBjNDA3ZmU5NWYzOWNjYzM3ZmE1NWJlYWI5ZjdkNjFiNjYwNzA3OWNkMTkwZjU0ZjZjNWFmZTliODQ2NTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wl4yjdrM7Kri7X082pOMbQovaAtQB7gPpR5tWT5Ghf0) Does it work normally?

And the log is as follows:
(textdiffuser2) jovyan@nb-big-dz-mxfw-1-0:~/wrj/workspace/project/unilm/textdiffuser-2$ bash inference_textdiffuser2_t2i_full.sh 
/opt/conda/envs/textdiffuser2/lib/python3.8/site-packages/diffusers/utils/outputs.py:63: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  torch.utils._pytree._register_pytree_node(
GPU name: NVIDIA A100 80GB PCIe
Number of GPUs: 1
Namespace(cache_dir=None, cfg=7.5, checkpointing_steps=500, checkpoints_total_limit=5, coord_mode='ltrb', dataloader_num_workers=0, drop_caption=False, enable_xformers_memory_efficient_attention=True, granularity=128, hub_model_id=None, hub_token=None, input_file=None, input_format='prompt', input_prompt='the log for "ABC"', local_rank=-1, logging_dir='logs', m1_model_path='/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2_layout_planner', max_length=77, mixed_precision='fp16', output_dir='inference_results', pretrained_model_name_or_path='/home/jovyan/wrj/workspace/project/tools/stable-diffusion-v1-5', prompts_txt_file=None, push_to_hub=False, report_to='tensorboard', resolution=512, resume_from_checkpoint='/home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2-full-ft', revision=None, sample_steps=20, seed=43555, vis_num=16)
/opt/conda/envs/textdiffuser2/lib/python3.8/site-packages/accelerate/accelerator.py:401: UserWarning: `log_with=tensorboard` was passed but no supported trackers are currently installed.
  warnings.warn(f"`log_with={log_with}` was passed but no supported trackers are currently installed.")
Detected kernel version 5.4.160, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.
05/31/2024 09:18:32 - INFO - __main__ - Distributed environment: NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: fp16

***************
49408
51583
***************
{'scaling_factor', 'force_upcast'} was not found in config. Values will be initialized to default values.
{'addition_time_embed_dim', 'reverse_transformer_layers_per_block', 'transformer_layers_per_block', 'dropout', 'attention_type'} was not found in config. Values will be initialized to default values.
Resuming from checkpoint textdiffuser2-full-ft
05/31/2024 09:18:46 - INFO - accelerate.accelerator - Loading states from /home/jovyan/wrj/workspace/project/unilm/textdiffuser-2/ckpt/JingyeChen22/textdiffuser2-full-ft
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - All model weights loaded successfully
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - All optimizer states loaded successfully
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - All scheduler states loaded successfully
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - All dataloader sampler states loaded successfully
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - GradScaler state loaded successfully
05/31/2024 09:18:46 - INFO - accelerate.checkpointing - Could not load random states
05/31/2024 09:18:46 - INFO - accelerate.accelerator - Loading in 0 custom states
detect existing output_dir, removing the contained jpg/txt files ...
rm: cannot remove 'inference_results/*.jpg': No such file or directory
rm: cannot remove 'inference_results/*.txt': No such file or directory
Loading checkpoint shards: 100%|██████████████████| 3/3 [00:25<00:00,  8.50s/it]
there are 1 samples for generation
[Human]
Given a prompt that will be used to generate an image, plan the layout of visual text for the image. The size of the image is 128x128. Therefore, all properties of the positions should not exceed 128, including the coordinates of top, left, right, and bottom. All keywords are included in the caption. You dont need to specify the details of font styles. At each line, the format should be keyword left, top, right, bottom. So let us begin. Prompt: the log for "ABC"
[Assistant]
ABC 22,24,114,79

the number of samples: 1
user_prompt the log for "ABC"
current_ocr ['ABC 22,24,114,79', '']
/opt/conda/envs/textdiffuser2/lib/python3.8/site-packages/diffusers/utils/outputs.py:63: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  torch.utils._pytree._register_pytree_node(
{'clip_sample_range', 'prediction_type', 'timestep_spacing', 'thresholding', 'variance_type', 'dynamic_thresholding_ratio', 'sample_max_value'} was not found in config. Values will be initialized to default values.
100%|███████████████████████████████████████████| 20/20 [00:07<00:00,  2.76it/s]

**Comment by ssotobayashi-m** [comment]
Thank you for the excellent research! I also tried similar experiments, and it seems that `inference_textdiffuser2_t2i_full.py` actually generates images that aren't as expected. On the other hand, when using `demo_textdiffuser2_t2i_full.py`, the images were generated as expected without any issues. This Python script appears to be essentially the same as the `app.py` deployed on Hugging Face.

I found the Gradio interface a bit inconvenient, so I modified `demo_textdiffuser2_t2i_full.py` to turn it into a command-line tool. Additionally, since `textdiffuser2_layout_planner` takes a bit of time to load (around 3 min), I converted it to the safetensors format, which allowed it to load much faster.

For reference, I've attached the environment information of the setup I experimented with, along with an example image generated using the prompt `--input_prompt 'A picture of a bruised apple with the text apples are good for you'` in `demo_textdiffuser2_t2i_full.py`.

- Python 3.10.18
- PyTorch 2.0.1

[requirements.txt]
```
numpy<2
diffusers==0.24.0
accelerate==0.22.0
datasets==2.11.0
transformers==4.28.1
fschat==0.2.26
pillow==10.1.0
sentencepiece==0.2.0
opencv-python-headless<4.10
xformers==0.0.20
protobuf==6.31.1
huggingface_hub==0.24.7
gradio==3.50.2
```

<img src="https://github.com/user-attachments/assets/64bd9663-ffa8-47ff-b20c-cf22fe785b9e" width="250px">
